docs/MinecraftApi/@minecraft/server/runCommandAsync.md

```diff
@@ -1,105 +1,116 @@
 # runCommandAsync
 
-Runs a particular command asynchronously from the context of the broader dimension.  Note that there is a maximum queue of 128 asynchronous commands that can be run in a given tick.
+Runs a particular command asynchronously from the context of the broader dimension.
+Note that there is a maximum queue of 128 asynchronous commands that can be run in a given tick.
 
-Normally we recommend avoiding using commands, however, the following command features are not implemented in scripting API (as of 1.19.50).
+Returns a `Promise<CommandResult>` . Throws an error **synchronously** if the queue is full.
 
-## @s[hasitem=]
+## Commands you have to use
 
-1. Can't access enderchest.
+Normally we recommend avoiding using commands,
+however, the following command features are not implemented in scripting API (as of 1.19.50).
+
+### @s
+
+1. Can't get entity by unique id.
+
+### @s[hasitem=]
+
+1. Can't access ender chest.
 2. Can't access equipments.
 
-## /replaceitem
+### /replaceitem
 
-1. Can't access enderchest.
+1. Can't access ender chest.
 2. Can't access equipments.
 
-## /clear
+### /clear
 
-1. Can't access enderchest.
+1. Can't access ender chest.
 2. Can't access equipments.
 
-## /tickingarea
+### /tickingarea
 
 1. Can't access ticking areas.
 
-## /kick
+### /kick
 
 1. Can't kick player.
 
-## /setblock
+### /setblock
 
 1. `/setblock ... destroy`
 
-## /fill
+### /fill
 
 1. `/fill` is fast.
 
-## /ability
+### /ability
 
-1. You can't set abilities for each players
-2. You can't read player's abilites
+1. You can't set abilities for each player.
+2. You can't read player's abilites.
 
-## /damage
+### /damage
 
-1. There's no way to give damage to players & entities
-2. Opening GUI with custom command will not possible (unless there's another way) because it's need damage for close the chat UI
+1. There's no way to give damage to players & entities.
+2. Opening GUI with custom command will not possible (unless there's another way) because it's need damage for close the chat UI.
```

Comment:
Can we use /scriptevent to overcome this limitation?
*or [force-show](https://github.com/JaylyDev/ScriptAPI/tree/main/scripts/force-show)*

Reply:
I don't know. I just copied it from the old file.
`forceShow` can't close chat, it only waits.
And I don't understand what here is related to `/scriptevent`.

Reply:
runCommandAsync(scriptevent) doesn't close chat ui. The client doesn't even know about that, I guess.
If the script's cmd system relies on `/scriptevent` and can't be used with chat msg then yes it can close the chat. But won't that result in user's complain?